src/main/java/ladder/domain/ExecutionResultMap.java

```diff
@@ -0,0 +1,50 @@
+package ladder.domain;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collector;
+import java.util.stream.IntStream;
+
+public class ExecutionResultMap {
+    private Map<String, String> executionResultMap;
+
+    private ExecutionResultMap() {
+        this(new LinkedHashMap<>());
+    }
+
+    private ExecutionResultMap(final Map<String, String> executionResultMap) {
+        this.executionResultMap = executionResultMap;
+    }
+
+    public static ExecutionResultMap of(Persons persons, LadderExecutionResults ladderExecutionResults) {
+        return IntStream.range(0, persons.getCountOfPerson())
+                .boxed()
+                .collect(Collector.of(
+                        ExecutionResultMap::new,
+                        (executionResultMap, index) ->
+                                executionResultMap.add(
+                                        persons.getPerson(index),
+                                        ladderExecutionResults.getExecutionResult(index)),
```

Comment:
```suggestion
                        updateMap(persons, ladderExecutionResults),
```
람다식 내부에 구현이 들어가서 내용이 길어지만 간결함이 훼손됩니다. 절절하게 extract method 하여 가독성을 유지할 필요가 있습니다.